x-pack/plugins/enterprise_search/server/lib/ml/fetch_ml_models.ts

```diff
@@ -127,7 +163,21 @@ const mergeModel = (model: MlModel, models: MlModel[]) => {
   }
 };
 
+const isCompatiblePromotedModelId = (modelId: string) =>
+  [compatibleElserModelId, compatibleE5ModelId].includes(modelId);
+
+/**
+ * A model is supported if:
+ * - The inference type is supported, AND
+ * - The model is the compatible variant of ELSER/E5, or it's a 3rd party model
+ */
 const isSupportedModel = (modelConfig: MlTrainedModelConfig) =>
+  isSupportedInferenceType(modelConfig) &&
+  ((!modelConfig.model_id.startsWith(ELSER_MODEL_ID) &&
+    !modelConfig.model_id.startsWith(E5_MODEL_ID)) ||
+    isCompatiblePromotedModelId(modelConfig.model_id));
```

Comment:
This logic - to filter the non-recommended ELSER/E5 version - is based on a strong assumption that curanted models have the same ID prefix for each variant, and that prefix is the ID of the cross-platform variant. For example:  `.elser_model_2` -> `.elser_model_2_linux-x86_64`. This is currently true for the offered ELSER and E5 variants. If this changes in a future release, we need to revisit this logic.
cc @darnautov

Reply:
@TattdCodeMonkey Correct, it's not used anywhere else for now, only for new pipeline creation (I'll double check). But it's a good call, we need to be careful when we consolidate the two pieces of list fetching logic.

Reply:
What was the reason for not utilizing the ML shared service to retrieve the list of supported models?

Reply:
@darnautov We call `getTrainedModels()` to fetch the full list and cross-check it with `getCuratedModelConfig()` for ELSER/E5. Is there a different API we should use?

Reply:
Just spotted it the `fetchCompatiblePromotedModelIds`! Mostly wondering if this check for supported is redundant, as it also involves maintaining the model constants in you plugin.